controllers/state_of_the_world.go

```diff
@@ -0,0 +1,149 @@
+package controllers
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	"github.com/kuadrant/policy-machinery/controller"
+	"github.com/kuadrant/policy-machinery/machinery"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	ctrlruntime "sigs.k8s.io/controller-runtime"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	kuadrantv1alpha1 "github.com/kuadrant/kuadrant-operator/api/v1alpha1"
+	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+	kuadrantv1beta2 "github.com/kuadrant/kuadrant-operator/api/v1beta2"
+)
+
+var kuadrantNamespace = "kuadrant-system"
```

Comment:
The kuadrant namespace is not hardcoded, it could be any.

Reply:
I agree, and raised this concern two weeks ago with out any feedback. The reason I have done this is for the topology configmap.  I am not happy watching every namespace on the cluster for configmaps, it feels like an over reach of access. But at the same time at initialization time of the controller I do not know how to get the namespace it is deployed into. If you have  suggestions on how to get the namespace dynamically during start up I will be more than happy to redo this.

Reply:
If we're filtering the ConfigMaps by label, watching all namespaces should be fine. Not ideal, I agree, but good enough for now.

Reply:
You can easily get the kuadrant namespace reading kuadrant CRs, pick one with a valid status (the remaining ones should not exist or they are irrelevant) and read the namespace.

Reply:
A topology may exist before the first Kuadrant CR is created. The "topology" ConfigMap still needs to be reconciled nonetheless.